bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/models/form/Modal.java

```diff
@@ -0,0 +1,30 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2024 Adobe
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+
+package com.adobe.cq.forms.core.components.models.form;
+
+import org.osgi.annotation.versioning.ConsumerType;
+
+/**
+ * Defines the form {@code Modal} Sling Model used for the
+ * {@code /apps/core/fd/components/form/modal/v1/modal} component.
+ *
+ * @since com.adobe.cq.forms.core.components.models.form 4.7.0
+ */
+@ConsumerType
+public interface Modal extends Container, ContainerConstraint, Base {
```

Comment:
This class is also not required !!

Reply:
Ok. The only way it is different from panel is that unlike, it doesn't have repeatability, we have left that for later.

Also, there would not be a dialog inside a dialog.